### PR TITLE
Handle invalid times in sync integration

### LINKED ISSUE #923: Fatal error
Not sure what happened - the dj demo site was not running, with only this hint in the log

```
2022-12-08 13:04:22.414718 ERROR omsupply_service::omsupply_service - panic occurred PanicInfo { payload: Any { .. }, message: Some(invalid time), location: Location { file: "C:\\Users\\Administrator\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\chrono-0.4.19\\src\\naive\\time.rs", line: 412, col: 9 }, can_unwind: true }  windows\src/windows.rs:66
```

have uploaded the postgres db and the datafile to [data share](https://drive.google.com/drive/folders/1tzCrTy65oZ_dm8SB2kbZX1N9q7YPCTw1?usp=share_link)

have stopped mSupply on the demo server, so that the demo runs. otherwise, it crashes on start up.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #923 
Removes the immediate problem. Doesn't offer an answer of why the values were invalid - all of the values checked were within the expected range for number of seconds and yet the fn call panicked still 🤷 timezone perhaps?

### CODE REVIEW

File: server/service/src/sync/synchroniser.rs [hunk 1]
```diff
@@ -208,13 +208,13 @@ pub fn integrate_and_translate_sync_buffer(
             let sync_buffer = SyncBuffer::new(connection);
             let translation_and_integration =
                 TranslationAndIntegration::new(connection, &sync_buffer);
-            // Translate and integrate upserts (ordered by referencial database constraints)
+            // Translate and integrate upserts (ordered by referential database constraints)
```
Comment (mark-prins) [inline server/service/src/sync/synchroniser.rs:211]: sorry, just being a pedant

File: server/service/src/sync/translations/requisition.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use util::constants::NUMBER_OF_DAYS_IN_A_MONTH;
 use crate::sync::{
     api::RemoteSyncRecordV5,
     sync_serde::{
-        date_and_time_to_datatime, date_from_date_time, date_option_to_isostring,
+        date_and_time_to_datetime, date_from_date_time, date_option_to_isostring,
```
Comment (mark-prins) [inline server/service/src/sync/translations/requisition.rs:14]: I prefer the original - in my head you say it with a terrible Italian accent 😉

**Comment by Chris-Petty** [comment]
Seems like this would fix immediate issue as you say... but there is something funky happening earlier in the system.

1. The UUID format suggests the invoice originates from open mSupply
2. Some store has sent an invoice to the store on the site you are syncing, or you are resyncing the device after using the store in omSupply
3. So somehow `NaiveTime` -> 4D `ALPHA 255` -> integration buffer JSON with 3 billion for a value that shouldn't exceed 86400. Maybe the serialisation isn't doing what we expect? Something funky going from u32 to JSON `number`?

**Comment by Chris-Petty** [comment]
I just made an invoice from omSupply to mSupply, in the mSupply buffer JSON it looks like this:
image

When in integrated it gets:
image
(This surprised me. I assumed it'd send a number of seconds since midnight)

Which when displayed in 4D is all good (if only a slight rounding error):
image

No answers, just good to see that in testing it works as intended... weird bug!

**Review by Chris-Petty — APPROVED** [review]
LGTM! I was able to sync the provided database with `Kalaf` site with no integration errors (that matter anyway). 

Could perhaps add logging? Don't know if it's any help without syncing (non-user related) logs anyway.

Still curious where the wacky numbers originate from!

File: server/service/src/sync/sync_serde.rs [hunk 3]
```diff
@@ -59,5 +59,6 @@ where
 /// change on our side.
 pub fn naive_time<'de, D: Deserializer<'de>>(d: D) -> Result<NaiveTime, D::Error> {
     let secs = u32::deserialize(d)?;
-    Ok(NaiveTime::from_num_seconds_from_midnight(secs, 0))
+    Ok(NaiveTime::from_num_seconds_from_midnight_opt(secs, 0)
+        .unwrap_or(NaiveTime::from_hms(0, 0, 0)))
```
Comment (clemens-msupply) [inline server/service/src/sync/sync_serde.rs:63]: What was the problem with the input data? number of seconds > 24h? should we fix it manually or is the input just broken?

**Review by clemens-msupply — APPROVED** [review]
Looks good, thanks!

File: server/service/src/sync/sync_serde.rs (hunk 3 above)
Comment (clemens-msupply) [reply server/service/src/sync/sync_serde.rs:63]: Nit: add comment in the code about it

**Pull request merged**